Feat stream generic

JhouXerox opened the pull request:
update stream.go to generic.

But here's a problem.

Merge and Group and Split    should return Stream[[]T], but it seems generic feature is not support to do this for now.

so for current It returns a new type ArrayStream.Already Implemented ForEach,ForAll,Done

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
Merge branch 'master' into feat_stream_generic

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
Merge branch 'master' into feat_stream_generic

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
update stream func to generic type.

JhouXerox committed:
Merge branch 'master' into feat_stream_generic

JhouXerox committed:
Merge remote-tracking branch 'origin/feat_stream_generic' into feat_stream_generic

JhouXerox committed:
deal with review dog error

JhouXerox committed:
Merge branch 'master' into feat_stream_generic

JhouXerox committed:
add more tests

kevwan commented on core/fx/stream.go:
```diff
@@ -279,37 +298,37 @@ func (s Stream) Head(n int64) Stream {
 // Map converts each item to another corresponding item, which means it's a 1:1 model.
-func (s Stream) Map(fn MapFunc, opts ...Option) Stream {
-	return s.Walk(func(item any, pipe chan<- any) {
+func (s Stream[T]) Map(fn MapFunc[T], opts ...Option) Stream[T] {
```
Map should convert items from T to another type.

Impossible for golang at the moment.

Reduce with the same problem.

JhouXerox replied:
Can't Map directly under Stream[T]. It returns Stream[any] for now.
For now it can only be used by general func, not method of instance.
so I provide a public func from stream package. See TestMapNew

JhouXerox committed:
Merge branch 'master' into feat_stream_generic

JhouXerox committed:
for now Map func can only return any because of the golang generic feature.
provide new Map func to pass by generic,it can't be invoked by instance. you can find in TestMapNew of stream_test.go , line 328.

JhouXerox committed:
Merge branch 'master' into feat_stream_generic

JhouXerox committed:
Merge branch 'master' into feat_stream_generic

kevwan committed:
Merge branch 'master' into feat_stream_generic
